Tests/ComposedUITests/ChangesReducerTests.swift

```diff
@@ -0,0 +1,259 @@
+        changesReducer.insertGroups([0, 1, 2, 3, 4])
+        changesReducer.removeGroups([2])
+
+        let changeset = changesReducer.endUpdating()
+
+        XCTAssertNotNil(changeset)
+
+        XCTAssertEqual(changeset!.groupsInserted, [0, 1, 2, 3])
+        XCTAssertTrue(changeset!.groupsRemoved.isEmpty)
+        XCTAssertTrue(changeset!.groupsUpdated.isEmpty)
+        XCTAssertTrue(changeset!.elementsRemoved.isEmpty)
+        XCTAssertTrue(changeset!.elementsInserted.isEmpty)
+        XCTAssertTrue(changeset!.elementsUpdated.isEmpty)
+        XCTAssertTrue(changeset!.elementsMoved.isEmpty)
+    }
+
+    func testMoveElement() {
+        var changesReducer = ChangesReducer()
+        changesReducer.beginUpdating()
+
+        changesReducer.moveElements([(from: IndexPath(row: 0, section: 0), to: IndexPath(row: 1, section: 0))])
+
+        let changeset = changesReducer.endUpdating()
+
+        XCTAssertNotNil(changeset)
+
+        XCTAssertEqual(
+            changeset!.elementsMoved,
+            [
+                Changeset.Move(
+                    from: IndexPath(row: 0, section: 0),
+                    to: IndexPath(row: 1, section: 0)
+                )
+            ]
+        )
+        XCTAssertTrue(changeset!.elementsRemoved.isEmpty)
+        XCTAssertTrue(changeset!.elementsInserted.isEmpty)
+        XCTAssertTrue(changeset!.elementsUpdated.isEmpty)
+        XCTAssertTrue(changeset!.groupsRemoved.isEmpty)
+        XCTAssertTrue(changeset!.groupsRemoved.isEmpty)
+        XCTAssertTrue(changeset!.groupsUpdated.isEmpty)
+    }
+
+    func testMoveElementThenRemoveElementBeforeMovedElement() {
+        var changesReducer = ChangesReducer()
+        changesReducer.beginUpdating()
+
+        /**
+         This is testing:
+
+         - A
+         - B
+         - C
+         - D
+
+         # Swap C and D
+
+         - A
+         - B
+         - D
+         - C
+
+         # Delete A
+
+         - B
+         - D
+         - C
+         */
+
+        changesReducer.moveElements([(from: IndexPath(row: 2, section: 0), to: IndexPath(row: 3, section: 0))])
+        changesReducer.removeElements(at: [IndexPath(row: 0, section: 0)])
+
+        let changeset = changesReducer.endUpdating()
+
+        XCTAssertNotNil(changeset)
+
+        XCTAssertEqual(
+            changeset!.elementsMoved,
+            [
+                Changeset.Move(
+                    from: IndexPath(row: 1, section: 0),
+                    to: IndexPath(row: 2, section: 0)
+                )
+            ]
+        )
+        XCTAssertEqual(changeset!.elementsRemoved, [IndexPath(row: 0, section: 0)])
+        XCTAssertTrue(changeset!.elementsInserted.isEmpty)
+        XCTAssertTrue(changeset!.elementsUpdated.isEmpty)
+        XCTAssertTrue(changeset!.groupsRemoved.isEmpty)
+        XCTAssertTrue(changeset!.groupsRemoved.isEmpty)
+        XCTAssertTrue(changeset!.groupsUpdated.isEmpty)
+    }
+
+    func testRemoveAMovedIndex() {
```

Comment:
@shaps80 This test, along with `testMoveElementAtSameIndexAsRemove` below, are currently failing but I'm not 100% sure about the tests themselves.

Can you confirm that the test looks correct? e.g. this should produce removes and inserts, not a remove and moves?